### PR TITLE
Adding documentation for GKE, AKS and EKS deployment for AKO.

### PR DESCRIPTION
**Pull request by parimanur**

### CODE REVIEW

File: docs/AKO/public_clouds.md [hunk 1]
```diff
@@ -91,24 +91,67 @@ Following lists the Day 0 preparation work required to set up AKO in Azure
...
+    * Run `configure serviceenginegroup seg1`
+    * Override the `se_subnet_id` in the SE Group by running below commands.
+        * `se_subnet_id subnet1`
+        * `save`
+    * Similarly override the `se_subnet_id` for the `cluster2` .
+
+## AKO deployment in GKE, AKS and EKS.
+
+In all of the deployments the cluster is deployed such that Pod IP addresses are natively routable. Set `AKOSettings.disableStaticRouteSync` to `true` in AKO for each of the deployment.
+
```
Comment (anish-avi) [inline docs/AKO/public_clouds.md:119]: We should add a comment saying that CNI plugin should be set as empty if they are not using any CNI on the cluster.

File: docs/AKO/public_clouds.md [hunk 2]
```diff
@@ -91,24 +91,67 @@ Following lists the Day 0 preparation work required to set up AKO in Azure
...
+Amazon Elastic Kubernetes Service (EKS) is a managed Kubernetes service that makes it easy for you to run Kubernetes on AWS and on-premises.
+
+Below points list down the EKS deployment options as well as the AKO deployment details.
+
+* EKS cluster is required to have at least two subnets configured on it from two different zones in same VPC.
+    * One of these subnets will be used to bring up the Avi service engines.
+* Amazon EKS works with the Project Calico network policy engine to provide fine grained networking policies for your Kubernetes workloads.
+    * Set `AKOSettings.cniPlugin` to calico in AKO. (optional)
+* By default EKS cluster does not have any nodes configured on it. A new nodegroup/autoscaling group needs to be created in the same subnet as EKS and associate it to the cluster
+* Controller is brought up on an altogether different subnet.
```
Comment (anish-avi) [inline docs/AKO/public_clouds.md:146]: Change this to "Controller can brought up on an altogether different subnet." It's not required to be in different subnet

File: docs/AKO/public_clouds.md [hunk 3]
```diff
@@ -91,24 +91,67 @@ Following lists the Day 0 preparation work required to set up AKO in Azure
...
+* Pods on the GKE cluster should be reachable to the AVI Controller.
+* Pod IP addresses should be natively routable within the cluster's VPC network and other VPC networks connected to it by VPC Network Peering.
+    * [VPC-native clusters](https://cloud.google.com/kubernetes-engine/docs/concepts/alias-ips#overview) in GCP supports this by default and this should be enabled for the cluster creation.
+* GKE cluster needs to have below changes to the non default parameters.
+    * `Enable HTTP load balancing` should be unchecked as we do not want to use the Google Cloud Load Balancer with Kubernetes Ingress.
+    \*![Alt text](images/public_cloud/gke_networking.png?raw=true)
+
+    *    `Enable network policy`  (optional) can be selected to install Calico CNI, GKE has built-in support for Calico.
+
+### AKO deployment in EKS
```
Comment (apalsule) [inline docs/AKO/public_clouds.md:135]: can you add a note that, AKO does not yet support multi-az VIPs for AWS. All the VIPs will be in a single subnet and single AZ.

File: docs/AKO/public_clouds.md (hunk 1 above)
Comment (parimanur) [reply docs/AKO/public_clouds.md:119]: Yes, that is by default, Also in the individual implementation i have marked them as optional.

File: docs/AKO/public_clouds.md (hunk 3 above)
Comment (parimanur) [reply docs/AKO/public_clouds.md:135]: Done.

File: docs/AKO/public_clouds.md (hunk 2 above)
Comment (parimanur) [reply docs/AKO/public_clouds.md]: Done! Thanks!

File: docs/AKO/public_clouds.md [hunk 4]
```diff
@@ -91,24 +91,69 @@ Following lists the Day 0 preparation work required to set up AKO in Azure
...
+
+In all of the deployments the cluster is deployed such that Pod IP addresses are natively routable. Set `AKOSettings.disableStaticRouteSync` to `true` in AKO for each of the deployment.
+
+### AKO deployment in GKE
+
+Google Kubernetes Engine (GKE) provides a managed environment for deploying, managing, and scaling your containerized applications using Google infrastructure.
+
+Below points list down the GKE deployment options as well as the AKO deployment details.
+
+* Pods on the GKE cluster should be reachable to the AVI Controller.
```
Comment (sudswasavi) [inline docs/AKO/public_clouds.md:126]: Should it be routable to the controller or should be routable to the Service Engines aka the VIP network?

**Review by sudswasavi — COMMENTED** [review]
One minor clarification point.

File: docs/AKO/public_clouds.md (hunk 4 above)
Comment (parimanur) [reply docs/AKO/public_clouds.md:126]: I have added this as AKO should reach the controller.

**Review by sudswasavi — APPROVED** [review]

**Pull request merged**